src/tools/manifest2proto.ts

```diff
@@ -135,11 +135,16 @@ export async function typeToProtoPayload(type: Type) {
   }
   type = type.resolvedType();
   switch (type.tag) {
-    case 'Entity': return {
-      entity: {
-        schema: await schemaToProtoPayload(type.getEntitySchema())
+    case 'Entity':
+      const entity = {
+        entity: {
+          schema: await schemaToProtoPayload(type.getEntitySchema()),
+        }
+      };
+      if (type.getEntitySchema().refinement) {
+        entity['refinement'] = type.getEntitySchema().refinement.toProto();
```

Comment:
I made a decision to add the refinement to the *entity* since `Type` is recursively defined with entity as a base case.

Reply:
This could be an argument to associate the refinement with the *schema*. 

More discussion about this here: #5257